File: flutter_app/lib/screens/dashboard_screen.dart
import 'package:flutter/material.dart';
import '../models/sale.dart';
import '../services/api_service.dart';

class DashboardScreen extends StatefulWidget {
  const DashboardScreen({super.key});

  @override
  State<DashboardScreen> createState() => _DashboardScreenState();
}

class _DashboardScreenState extends State<DashboardScreen> {
  final _api = ApiService();
  DashboardStats _stats = DashboardStats();
  bool _loading = true;

  @override
  void initState() {
    super.initState();
    _refresh();
  }

  Future<void> _refresh() async {
    setState(() => _loading = true);
    try {
      _stats = await _api.getDashboardStats();
    } catch (_) {}
    setState(() => _loading = false);
  }

  @override
  Widget build(BuildContext context) {
    if (_loading) return const Center(child: CircularProgressIndicator());

    return RefreshIndicator(
      onRefresh: _refresh,
      child: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          Text('Overview', style: Theme.of(context).textTheme.headlineSmall),
          const SizedBox(height: 16),
          Row(children: [
            Expanded(child: _StatCard(label: 'Products', value: '${_stats.totalProducts}')),
            const SizedBox(width: 12),
            Expanded(child: _StatCard(
              label: 'Low Stock',
              value: '${_stats.lowStockCount}',
              isWarning: _stats.lowStockCount > 0,
            )),
          ]),
          const SizedBox(height: 12),
          Row(children: [
            Expanded(child: _StatCard(label: "Today's Sales", value: '${_stats.todaySales}')),
            const SizedBox(width: 12),
            Expanded(child: _StatCard(label: 'Revenue', value: '¥${_stats.todayRevenue.toStringAsFixed(0)}')),
          ]),
        ],
      ),
    );
  }
}

class _StatCard extends StatelessWidget {
  final String label;
  final String value;
  final bool isWarning;

  const _StatCard({required this.label, required this.value, this.isWarning = false});

  @override
  Widget build(BuildContext context) {
    return Card(
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(label, style: Theme.of(context).textTheme.labelMedium),
            const SizedBox(height: 4),
            Text(value,
              style: Theme.of(context).textTheme.headlineMedium?.copyWith(
                fontWeight: FontWeight.bold,
                color: isWarning ? Theme.of(context).colorScheme.error : null,
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: android/app/src/main/java/com/pos/inventory/ui/screens/DashboardScreen.kt
package com.pos.inventory.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.pos.inventory.data.api.ApiService
import com.pos.inventory.data.api.dto.DashboardStatsDto
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class DashboardViewModel @Inject constructor(
    private val api: ApiService,
) : ViewModel() {
    private val _stats = MutableStateFlow(DashboardStatsDto())
    val stats = _stats.asStateFlow()

    private val _loading = MutableStateFlow(true)
    val loading = _loading.asStateFlow()

    init { refresh() }

    fun refresh() {
        viewModelScope.launch {
            _loading.value = true
            try {
                _stats.value = api.getDashboardStats()
            } catch (_: Exception) { }
            _loading.value = false
        }
    }
}

@Composable
fun DashboardScreen(viewModel: DashboardViewModel = hiltViewModel()) {
    val stats by viewModel.stats.collectAsState()
    val loading by viewModel.loading.collectAsState()

    if (loading) {
        Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            CircularProgressIndicator()
        }
    } else {
        Column(
            modifier = Modifier.fillMaxSize().padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            Text("Overview", style = MaterialTheme.typography.headlineSmall)

            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                StatCard("Products", "${stats.totalProducts}", Modifier.weight(1f))
                StatCard("Low Stock", "${stats.lowStockCount}", Modifier.weight(1f),
                    isWarning = stats.lowStockCount > 0)
            }
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                StatCard("Today's Sales", "${stats.todaySales}", Modifier.weight(1f))
                StatCard("Revenue", "¥%.0f".format(stats.todayRevenue), Modifier.weight(1f))
            }
        }
    }
}

@Composable
private fun StatCard(
    label: String,
    value: String,
    modifier: Modifier = Modifier,
    isWarning: Boolean = false,
) {
    Card(modifier = modifier) {
        Column(Modifier.padding(16.dp)) {
            Text(label, style = MaterialTheme.typography.labelMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant)
            Spacer(Modifier.height(4.dp))
            Text(value, style = MaterialTheme.typography.headlineMedium,
                fontWeight = FontWeight.Bold,
                color = if (isWarning) MaterialTheme.colorScheme.error
                        else MaterialTheme.colorScheme.onSurface)
        }
    }
}
